architecture.md

```diff
@@ -0,0 +1,91 @@
+One common optional plugin is the sensors plugin.
+Both optional and default plugins can be removed or added at any time; Ignition Gazebo will continue to run with limited functionality.
+These demos on [Server Configuration](https://ignitionrobotics.org/api/gazebo/4.3/server_config.html) and [GUI Configuration](https://ignitionrobotics.org/api/gazebo/4.3/gui_config.html) showcase that functionality.
+
+The simulation process is depicted in the diagram below, and further explained in the Backend and Frontend process sections that follow.
+
+![Ignition Gazebo architecture diagram](images/IgnitonGazeboArchicteture.png)
+
+## Backend server process
+
+Ignition Gazebo is responsible for loading plugins in the backend, referred to as systems.
+The server runs an entity-component system architecture (see [Ignition Gazebo terminology](https://ignitionrobotics.org/api/gazebo/4.2/terminology.html)).
+The backend will usually have multiple systems responsible for everything in the simulation – computing physics, recording logs, receiving user commands, etc.
+
+Systems act on entities and components of those entities.
+An entity is anything in the simulation scene (a model, link, light, actor, etc.), and components are their characteristics (pose, name, geometry, etc.).
+Server plugins have access to all entities and what components they have, and make modifications to those components.
+For example, a user could write a system that applies force to an entity by setting a component on that entity, and the physics system would pick that up and react by applying the force to make the entity actually move.
+The modification of entities and components is how server plugins communicate with each other.
+Other Ignition Libraries, however, don’t know about the entities and components.
+
+There is a loop running in the backend that runs the systems, where some systems are proposing changes to entities and their components, and other systems are handling and applying those changes.
+This is called the “simulation loop”.
+The Entity Component Manager (ECM) in the backend provides the functionality for the actual querying and updating of the entities and components.
+
+Physics, user commands, and scene broadcaster are all systems launched by default in the backend.
+As mentioned earlier, however, even default systems can be added to or removed from this loop.
+For any other functionality, like sensor data processing for example, an additional system would have to be loaded.
+For example, if you need to generate sensor data that utilizes rendering sensors, you would need to load the sensors system.
+The visualization of that data, however, is left up to the client side plugins.
+
+## Communication process
+
+Any information that crosses the process boundary (whether back-to-front with scene broadcaster or front-to-back with user commands) has to go through [Ignition Transport](https://ignitionrobotics.org/libs/transport), the communication library.
```

Comment:
I think that this is the first time in the tutorial the term "process boundary" is used. Should we provide a definition?